cpp/src/arrow/csv/writer.cc

```diff
@@ -571,34 +556,39 @@ class CSVWriterImpl : public ipc::RecordBatchWriter {
           column_populators_[col]->UpdateRowLengths(*batch.column(col), offsets_.data()));
     }
     // Calculate cumulative offsets for each row (including delimiters).
-    // ',' * num_columns - 1(last column doesn't have ,) + eol
-    int32_t delimiters_length =
+    // - before conversion: offsets_[i] = length of i-th row
```

Comment:
```suggestion
    // - before conversion: offsets_[i] = length of (i-1)th row
```

Reply:
`offsets_[i]` is indeed the length of  `i-th` row before conversion, not `(i-1)th`.

Reply:
Oops, my bad.